features/admin.connections.v1/components/edit/forms/authenticators/push-authenticator-form.tsx

```diff
@@ -0,0 +1,491 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TestableComponentInterface } from "@wso2is/core/models";
+import { Field, Form } from "@wso2is/form";
+import { Code } from "@wso2is/react-components";
+import { FormValidation } from "@wso2is/validation";
+import isBoolean from "lodash-es/isBoolean";
+import isEmpty from "lodash-es/isEmpty";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { Label } from "semantic-ui-react";
+import { ConnectionUIConstants } from "../../../../constants/connection-ui-constants";
+import { LocalAuthenticatorConstants } from "../../../../constants/local-authenticator-constants";
+import {
+    CommonAuthenticatorFormFieldInterface,
+    CommonAuthenticatorFormFieldMetaInterface,
+    CommonAuthenticatorFormInitialValuesInterface,
+    CommonAuthenticatorFormMetaInterface,
+    CommonAuthenticatorFormPropertyInterface
+} from "../../../../models/authenticators";
+import {
+    CommonPluggableComponentMetaPropertyInterface,
+    CommonPluggableComponentPropertyInterface
+} from "../../../../models/connection";
+
+interface PushAuthenticatorFormPropsInterface extends TestableComponentInterface {
+    /**
+     * Push Authenticator Metadata.
+     */
+    metadata: CommonAuthenticatorFormMetaInterface;
+    /**
+     * Push Authenticator configured initial values.
+     */
+    initialValues: CommonAuthenticatorFormInitialValuesInterface;
+    /**
+     * Callback for form submit.
+     * @param values - Resolved Form Values.
+     */
+    onSubmit: (values: CommonAuthenticatorFormInitialValuesInterface) => void;
+    /**
+     * Is readonly.
+     */
+    readOnly?: boolean;
+    /**
+     * Flag to trigger form submit externally.
+     */
+    triggerSubmit: boolean;
+    /**
+     * Flag to enable/disable form submit button.
+     */
+    enableSubmitButton: boolean;
+    /**
+     * Flag to show/hide custom properties.
+     * @remarks Not implemented ATM. Do this when needed.
+     */
+    showCustomProperties: boolean;
+    /**
+     * Specifies if the form is submitting.
+     */
+    isSubmitting?: boolean;
+}
+
+/**
+ * Form initial values interface.
+ */
+interface PushAuthenticatorFormInitialValuesInterface {
+    /**
+     * Enable number challenge.
+     */
+    PUSH_EnableNumberChallenge: boolean;
+    /**
+     * Enable progressive enrollment.
+     */
+    PUSH_EnableProgressiveEnrollment: boolean;
+    /**
+     * Resend notification after time in seconds.
+     */
+    PUSH_ResendNotificationTime: number;
+    /**
+     * Resend notification max attempts.
+     */
+    PUSH_ResendNotificationMaxAttempts: number;
+}
+
+/**
+ * Form fields interface.
+ */
+interface PushAuthenticatorFormFieldsInterface {
+    /**
+     * Enable number challenge.
+     */
+    PUSH_EnableNumberChallenge: CommonAuthenticatorFormFieldInterface;
```

Comment:
Any specific reason why we appended PUSH as a prefix?

Reply:
@ZiyamSanthosh

Reply:
https://github.com/wso2/identity-apps/blob/f559f785926916c1fa0d37d1d4b2aaeab506907e/features/admin.connections.v1/components/forms/authenticators/sms-otp-authenticator-form.tsx#L89

This was the pattern used in other authenticators as well. I followed the existing pattern.